model/textparse/protobufparse.go

```diff
@@ -634,11 +640,17 @@ func formatOpenMetricsFloat(f float64, b []byte) string {
 	case math.IsInf(f, -1):
 		return "-Inf"
 	}
-	s := string(strconv.AppendFloat(b[:0], f, 'g', -1, 64))
-	if strings.ContainsAny(s, "e.") {
+	bp := floatFormatBufPool.Get().(*[]byte)
+	*bp = strconv.AppendFloat((*bp)[:0], f, 'g', -1, 64)
+	if bytes.ContainsAny(*bp, "e.") {
+		s := string(*bp)
+		floatFormatBufPool.Put(bp)
```

Comment:
Let's put the `Put` in a `defer` statement? I have vague memories that the performance overhead of that should be much smaller these days…

Being able to write `return string (*bp)` is totally worth it, IMHO. :)

Reply:
Yes, I wanted to test the version without defer but didn't restore.